File: src/main/java/pl/yalgrin/playnite/simplesync/mapper/ChangeMapper.java

package pl.yalgrin.playnite.simplesync.mapper;

import org.springframework.stereotype.Component;
import pl.yalgrin.playnite.simplesync.domain.Change;
import pl.yalgrin.playnite.simplesync.dto.ChangeDTO;

import java.time.Instant;

@Component
public class ChangeMapper {

    public Change toEntity(ChangeDTO dto) {
        Change entity = new Change();
        entity.setId(dto.getId());
        entity.setType(dto.getType());
        entity.setClientId(dto.getClientId());
        entity.setObjectId(dto.getObjectId());
        entity.setNotifyAll(dto.isForceFetch());
        entity.setCreatedAt(Instant.now());
        return entity;
    }

    public ChangeDTO toDTO(Change entity) {
        ChangeDTO dto = new ChangeDTO();
        dto.setId(entity.getId());
        dto.setType(entity.getType());
        dto.setClientId(entity.getClientId());
        dto.setObjectId(entity.getObjectId());
        dto.setForceFetch(entity.isNotifyAll());
        return dto;
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/change/mapper/ChangeMapper.kt
package pl.yalgrin.playnite.simplesync.change.mapper

import org.springframework.stereotype.Component
import pl.yalgrin.playnite.simplesync.change.domain.Change
import pl.yalgrin.playnite.simplesync.change.dto.ChangeDTO
import java.time.Instant

@Component
class ChangeMapper {
    fun toEntity(dto: ChangeDTO): Change {
        return Change(
            id = dto.id,
            type = dto.type,
            clientId = dto.clientId!!,
            objectId = dto.objectId,
            diffParentObjectId = dto.diffParentObjectId,
            notifyAll = dto.isForceFetch,
            createdAt = Instant.now()
        )
    }

    fun toDTO(entity: Change): ChangeDTO {
        return ChangeDTO(
            id = entity.id,
            type = entity.type,
            clientId = entity.clientId,
            objectId = entity.objectId,
            diffParentObjectId = entity.diffParentObjectId,
            isForceFetch = entity.notifyAll
        )
    }
}